File: chapter4/java/designpattern/src/main/java/com/designpattern/CreditCardPaymentStrategy.java
package com.designpattern;

/**
 * CreditCardPaymentStrategy as ConcreteStrategy class.
 * CreditCardPaymentStrategy implementation of PaymentStrategy interface with an CreditCard payment algorithm.
 *
 * @Author Bridget Wu
 */
public class CreditCardPaymentStrategy implements PaymentStrategy {
    /**
     * CreditCard Payment implementation detail
     *
     * @param price
     */
    @Override
    public void pay(int price) {
        System.out.println("Charged total prices $" + price + " from CreditCard.");
    }
}

File: chapter4/typescript/CreditCardPaymentStrategy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreditCardPaymentStrategy = void 0;
/**
 * CreditCardPaymentStrategy as ConcreteStrategy Classes.
 * CreditCardPaymentStrategy implementation of Strategy interface - PaymentStrategy implements an CreditCard payment algorithm.
 *
 * @Author Bridget Wu
 */
class CreditCardPaymentStrategy {
    /**
     * CreditCard Payment implementation detail
     *
     * @param price
     */
    pay(price) {
        console.log("Charged total prices $" + price + " from CreditCard.");
    }
}
exports.CreditCardPaymentStrategy = CreditCardPaymentStrategy;
